docs/codeql/ql-language-reference/ql-language-specification.rst

```diff
@@ -2040,7 +2042,7 @@ Stratification
 
 A QL program can be *stratified* to a sequence of *layers*. A layer is a set of predicates and types.
 
-A valid stratification must include each predicate and type in the QL program. It must not include any other predicates or types.
+A valid stratification must include each predicate and type in the QL program that is *fully instantiated*. It must not include any other predicates or types.
```

Comment:
I'm not too bothered but we also check the uninstantiated cases as well so we should include them. (on the other hand we would need to state that we ignore dispatch for parameters).

Reply:
Although we would also need to state evaluation only happens on *fully instantiated* predicates

Reply:
We report errors on uninstantiated cases as well, that is true. However, given an error-free QL program, the difference between adding in the uninstantiated cases in the stratification (and filtering out per layer afterwards) or not should not be observable. Therefore, I think we can get away with this here.